snakemake/satija/src/convert_h5ad_to_arrow.py

```diff
@@ -0,0 +1,42 @@
+import argparse
+
+from anndata import read_h5ad
+import pyarrow as pa
+import pandas as pd
+
+
+def h5ad_to_arrow(h5ad_file, arrow_file):
+    ann_data = read_h5ad(h5ad_file)
+    umap = ann_data.obsm['X_umap'].transpose()
+    leiden = ann_data.obs['leiden'].to_numpy().astype('uint8')
+    index = ann_data.obs.index
+
+    df = pd.DataFrame(
+        data={'umap_x': umap[0], 'umap_y': umap[1], 'leiden': leiden},
+        index=index
+    )
+    table = pa.Table.from_pandas(df)
+
+    writer = pa.RecordBatchFileWriter(arrow_file, table.schema)
+    writer.write(table)
+    writer.close()
```

Comment:
I think you use the context manager here, but you do close manually right after .. so probably just personal preference.

```suggestion
    with pa.RecordBatchFileWriter(arrow_file, table.schema) as writer:
        writer.write(table)
```